src/ProjectsUnderTest/Agoda.IoC.ProjectUnderTest.Valid2/ValidRegistrations2.cs

```diff
@@ -10,4 +10,23 @@ public class ReplaceServiceTwoWork : IReplaceService
 
         public string DoWork { get; set; } = nameof(ReplaceServiceTwoWork);
     }
+
+    public interface IServiceThatStartsUp
+    {
+        int Somedata { get; set; }
+    }
+
+    [RegisterSingleton(For = typeof(IServiceThatStartsUp))]
+    public class ServiceThatStartsUp : IServiceThatStartsUp, IStartupable
+    {
+        public ServiceThatStartsUp()
+        {
+            Somedata = 0;
+        }
+        public int Somedata { get; set; }
+        public void Start()
```

Comment:
Ideally, it would be nice if the startup process could run asynchronously. I would expect that most of the implementation will communicate with other services etc. therefore it might be better to start returning a Task from the start.